src/cpp/session/include/session/SessionAsyncRProcess.hpp

```diff
@@ -79,6 +79,8 @@ class AsyncRProcess :
 
    virtual void onCompleted(int exitStatus) = 0;
 
+   bool terminationRequested();
```

Comment:
Maybe this should be public along with the other methods that mutate / inspect state?